demos/dev_mode_key_provisioning/src/aws_key_provisioning_demo.c

```diff
@@ -0,0 +1,613 @@
+    CK_MECHANISM xMech = { 0 };
+    CK_BYTE xToBeSigned[ 256 ];
+    uint8_t ucTemp[ 64 ] = { 0 }; /* A temporary buffer for the pre-formatted signature. */
+    CK_ULONG xToBeSignedLen = sizeof( xToBeSigned );
+
+
+    /* Unreferenced parameters. */
+    ( void ) ( piRng );
+    ( void ) ( pvRng );
+    ( void ) ( xMdAlg );
+
+    /* Sanity check buffer length. */
+    if( xHashLen > sizeof( xToBeSigned ) )
+    {
+        xResult = CKR_ARGUMENTS_BAD;
+    }
+
+    xMech.mechanism = CKM_ECDSA;
+    memcpy( xToBeSigned, pucHash, xHashLen );
+    xToBeSignedLen = xHashLen;
+
+    if( 0 == xResult )
+    {
+        /* Use the PKCS#11 module to sign. */
+        xResult = pxGlobalFunctionList->C_SignInit( xGlobalSession,
+                                                    &xMech,
+                                                    xPrivateKeyHandle );
+    }
+
+    if( 0 == xResult )
+    {
+        *pxSigLen = sizeof( xToBeSigned );
+        xResult = pxGlobalFunctionList->C_Sign( ( CK_SESSION_HANDLE ) xGlobalSession,
+                                                xToBeSigned,
+                                                xToBeSignedLen,
+                                                pucSig,
+                                                ( CK_ULONG_PTR ) pxSigLen );
+    }
+
+    uint8_t * pucSigPtr;
+
+    /* PKCS #11 for P256 returns a 64-byte signature with 32 bytes for R and 32 bytes for S.
+     * This must be converted to an ASN1 encoded array. */
+    configASSERT( *pxSigLen == 64 );
+    memcpy( ucTemp, pucSig, *pxSigLen );
+
+    pucSig[ 0 ] = 0x30; /* Sequence. */
+    pucSig[ 1 ] = 0x44; /* The minimum length the signature could be. */
+    pucSig[ 2 ] = 0x02; /* Integer. */
+
+    if( ucTemp[ 0 ] & 0x80 )
+    {
+        pucSig[ 1 ]++;
+        pucSig[ 3 ] = 0x21;
+        pucSig[ 4 ] = 0x0;
+        memcpy( &pucSig[ 5 ], ucTemp, 32 );
+        pucSigPtr = pucSig + 33 + 4;
+    }
+    else
+    {
+        pucSig[ 3 ] = 0x20;
+        memcpy( &pucSig[ 4 ], ucTemp, 32 );
+        pucSigPtr = pucSig + 32 + 4;
+    }
+
+    pucSigPtr[ 0 ] = 0x02; /* Integer. */
+    pucSigPtr++;
+
+    if( ucTemp[ 32 ] & 0x80 )
+    {
+        pucSig[ 1 ]++;
+        pucSigPtr[ 0 ] = 0x21;
+        pucSigPtr[ 1 ] = 0x00;
+        pucSigPtr += 2;
+
+        memcpy( pucSigPtr, &ucTemp[ 32 ], 32 );
+    }
+    else
+    {
+        pucSigPtr[ 0 ] = 0x20;
+        pucSigPtr++;
+        memcpy( pucSigPtr, &ucTemp[ 32 ], 32 );
+    }
+
+    *pxSigLen = ( CK_ULONG ) pucSig[ 1 ] + 2;
+
+    if( xResult != 0 )
+    {
+        configPRINTF( ( "ERROR: Failure in signing callback: %d \r\n", xResult ) );
+        lFinalResult = TLS_ERROR_SIGN;
+    }
+
+    return lFinalResult;
+}
+
+/*-----------------------------------------------------------*/
+
+/* @brief Provision Device Using Device and CA Certificates
+ *
+ *
+ *	\return    Void
+ *  Prints error message upon failure, success message upon success
+ *
+ */
+static void vProvisionDevice( void )
```

Comment:
Consider renaming this function xProvisionDeviceForJITP so that it is not confused with the devmodekeyprovisioning function xProvisionDevice.

Same below for vDevModeDeviceKeyProvisioning ... let's highlight the difference between this function and the one that already exists with their functions
This makes me realize that perhaps I should rename vDevModeKeyProvisioning and  xProvisionDevice as well :)

Reply:
renamed!